File: results/java_codex/COUNT_PAIRS_WITH_GIVEN_SUM_1.java
int getPairsCount(int arr[], int n, int sum) {
        unordered_map<int, int> hm;
        for (int i = 0; i < n; i++) {
            if (hm.find(arr[i]) == hm.end())
                hm[arr[i]] = 0;
            hm[arr[i]] += 1;
        }
        int twice_count = 0;
        for (int i = 0; i < n; i++) {
            if (hm.find(sum - arr[i]) != hm.end())
                twice_count += hm[sum - arr[i]];
            if (sum - arr[i] == arr[i])
                twice_count--;
        }
        return twice_count / 2;
    }

File: TransCoder_tokenized_test_set_functions/python/COUNT_PAIRS_WITH_GIVEN_SUM_1.py
def getPairsCount ( arr, n, sum ) :
    m = [ 0 ] * 1000
    for i in range ( 0, n ) :
        m [ arr [ i ] ]
        m [ arr [ i ] ] += 1
    twice_count = 0
    for i in range ( 0, n ) :
        twice_count += m [ sum - arr [ i ] ]
        if ( sum - arr [ i ] == arr [ i ] ) :
            twice_count -= 1
    return int ( twice_count / 2 )
